rad/src/reducers/hash_concatenate.rs

```diff
@@ -0,0 +1,62 @@
+use crate::{
+    error::RadError,
+    hash_functions::RadonHashFunctions,
+    operators::bytes as bytes_operators,
+    reducers::RadonReducers,
+    types::{array::RadonArray, bytes::RadonBytes, integer::RadonInteger, RadonType, RadonTypes},
+};
+use serde_cbor::value::Value;
+
+/// Reducer that concatenate all the RadonTypes and hash
+#[allow(clippy::cast_precision_loss)]
+pub fn hash_concatenate(input: &RadonArray) -> Result<RadonTypes, RadError> {
+    let value = input.value();
+
+    match value.first() {
+        None => Ok(RadonTypes::from(RadonInteger::from(0))),
```

Comment:
Empty array case, return integer is wrong here. Either return error or return `sha256` of empty string, which is defined as `EMPTY_SHA256` in `witnet_crypto` crate.